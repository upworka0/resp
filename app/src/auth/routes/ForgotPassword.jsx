import React, { PureComponent } from 'react';
import {
  Card, Row, Col, CardBody, CardHeader,
} from 'reactstrap';

import { ForgotPasswordForm } from '../components';

export default class ForgotPassword extends PureComponent {
  constructor(props) {
    super(props);

    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(data) {
    // eslint-disable-next-line
    console.log('Reset Password', data);
  }

  render() {
    return (
      <section className="d-flex flex-grow-1 flex-column align-items-center justify-content-center">
        <div className="container">
          <Row noGutters>
            <Col
              md={{
                size: 8,
                offset: 2,
              }}
              lg={{
                size: 6,
                offset: 3,
              }}
            >
              <h1 className="text-center mb-3">Loji</h1>
              <Card className="border-0 shadow-z4">
                <CardHeader className="bg-primary p-1" />
                <CardBody>
                  <ForgotPasswordForm onSubmit={this.onSubmit} />
                </CardBody>
              </Card>
            </Col>
          </Row>
        </div>
      </section>
    );
  }
}
