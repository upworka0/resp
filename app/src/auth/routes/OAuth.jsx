import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Progress } from 'reactstrap';
import { Redirect } from 'react-router-dom';

export default class OAuth extends Component {
  state = {
    hasValidationBeenFailed: false,
    redirect: false,
  };

  componentDidMount() {
    const { validateToken, location } = this.props;

    const params = new URLSearchParams(location.search);
    const accessToken = params.get('access');
    const refreshToken = params.get('refresh');

    validateToken(accessToken, refreshToken)
      .then(() => {
        this.setState({ redirect: true });
      })
      .catch(() => {
        this.setState({ hasValidationBeenFailed: true });
      });
  }

  render() {
    const { hasValidationBeenFailed, redirect } = this.state;

    if (hasValidationBeenFailed) {
      return (<Redirect to="/signin" />);
    }

    if (redirect) {
      return (<Redirect to="/" />);
    }

    return (
      <section className="d-flex flex-column flex-grow-1 align-items-center justify-content-center">
        <div className="px-4 py-3 shadow-z3 rounded">
          <h5 className="text-center px-5">Verifying your token...</h5>
          <Progress animated value={100} />
        </div>
      </section>
    );
  }
}

OAuth.propTypes = {
  validateToken: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired,
};
