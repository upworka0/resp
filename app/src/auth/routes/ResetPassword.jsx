import React from 'react';
import {
  Form, FormGroup, Label, Input,
  Button, Card, Row, Col, CardBody,
  CardHeader, FormFeedback,
} from 'reactstrap';

export default function ResetPassword() {
  return (
    <section className="d-flex flex-grow-1 flex-column align-items-center justify-content-center">
      <div className="container">
        <Row noGutters>
          <Col
            md={{
              size: 8,
              offset: 2,
            }}
            lg={{
              size: 6,
              offset: 3,
            }}
          >
            <h1 className="text-center mb-3">Loji</h1>
            <Card className="border-0 shadow-z4">
              <CardHeader className="bg-primary p-1" />
              <CardBody>
                <Form>
                  <FormGroup>
                    <Label htmlFor="loginInput">
                      New password
                    </Label>
                    <Input
                      type="email"
                      className="form-control"
                      id="loginInput"
                      placeholder="enter your email address"
                    />
                    <FormFeedback>Field is required</FormFeedback>
                  </FormGroup>
                  <FormGroup>
                    <Label htmlFor="loginInput">
                      Confirm new password
                    </Label>
                    <Input
                      type="email"
                      className="form-control"
                      id="loginInput"
                      placeholder="enter your email address"
                    />
                    <FormFeedback>Field is required</FormFeedback>
                  </FormGroup>
                  <div className="d-flex justify-content-center">
                    <Button type="submit" color="primary">Change Password</Button>
                  </div>
                </Form>
              </CardBody>
            </Card>
          </Col>
        </Row>
      </div>
    </section>
  );
}
