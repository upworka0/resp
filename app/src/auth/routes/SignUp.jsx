import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import {
  Card, Row, Col, CardBody, CardHeader,
} from 'reactstrap';

import { SignUpForm } from '../components';
import { createCustomer } from '../../store/auth/authActions';

export class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isAuthenticating: false,
      redirect: false,
    };

    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit({
    email, password, username, keepSignedIn,
  }) {
    const { signUp } = this.props;

    this.setState({ isAuthenticating: true }, () => {
      signUp({ email, password, username }, keepSignedIn)
        .then(() => {
          this.setState({ isAuthenticating: false, redirect: true });
        })
        .catch((error) => {
          console.log(error); // @TODO: display correct error message
          this.setState({ isAuthenticating: false });
        });
    });
  }

  render() {
    const { isAuthenticating, redirect } = this.state;
    const { from } = this.props;

    if (redirect) {
      return <Redirect to={from} />;
    }

    return (
      <section className="d-flex flex-grow-1 flex-column align-items-center justify-content-center">
        <div className="container">
          <Row noGutters>
            <Col
              md={{
                size: 8,
                offset: 2,
              }}
              lg={{
                size: 6,
                offset: 3,
              }}
            >
              <h1 className="text-center mb-3">Loji</h1>
              <Card className="border-0 shadow-z4">
                <CardHeader className="bg-primary p-1" />
                <CardBody>
                  <SignUpForm
                    onSubmit={this.onSubmit}
                    isCustomerAuthenticating={isAuthenticating}
                  />
                </CardBody>
              </Card>
            </Col>
          </Row>
        </div>
      </section>
    );
  }
}

const mapActionsToProps = (dispatch) => ({
  signUp: (credentials, isKeepingSignedIn) => dispatch(createCustomer(credentials, isKeepingSignedIn)),
});

export default connect(null, mapActionsToProps)(SignUp);

SignUp.propTypes = {
  signUp: PropTypes.func.isRequired,
  from: PropTypes.string,
};

SignUp.defaultProps = {
  from: '/',
};
