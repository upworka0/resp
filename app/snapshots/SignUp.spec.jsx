import React from 'react';
import { shallow } from 'enzyme';

import { SignUp } from '../src/auth/routes/SignUp';

describe('SignUp UI', () => {
  const signUpMock = jest.fn();

  it('should render authentication form', () => {
    const wrapper = shallow(<SignUp signUp={signUpMock} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect once user is signed up', () => {
    const wrapper = shallow(<SignUp signUp={signUpMock} />);
    wrapper.setState({ redirect: true });
    expect(wrapper).toMatchSnapshot();
  });
});
