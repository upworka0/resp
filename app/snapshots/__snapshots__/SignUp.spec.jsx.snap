// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp UI should redirect once user is signed up 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`SignUp UI should render authentication form 1`] = `
<section
  className="d-flex flex-grow-1 flex-column align-items-center justify-content-center"
>
  <div
    className="container"
  >
    <Row
      noGutters={true}
      tag="div"
    >
      <Col
        lg={
          Object {
            "offset": 3,
            "size": 6,
          }
        }
        md={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h1
          className="text-center mb-3"
        >
          Loji
        </h1>
        <Card
          className="border-0 shadow-z4"
          tag="div"
        >
          <CardHeader
            className="bg-primary p-1"
            tag="div"
          />
          <CardBody
            tag="div"
          >
            <SignUpForm
              isCustomerAuthenticating={false}
              onSubmit={[Function]}
            />
          </CardBody>
        </Card>
      </Col>
    </Row>
  </div>
</section>
`;
